src/main/java/net/sf/jabref/MetaData.java

```diff
@@ -455,4 +457,20 @@ public Charset getEncoding() {
     public void setEncoding(Charset encoding) {
         this.encoding = Objects.requireNonNull(encoding);
     }
+
+    public void clearMetaData() {
+        metaData.clear();
+    }
+
+    public Map<String, List<String>> getMetaData() {
+        return metaData;
+    }
+
+    public void setMetaData(Map<String, List<String>> metaData) {
```

Comment:
Hmm, yea but in that case `DBMSSynchronizer` would have to be depentent on `BibDatabaseContext` which would create a bidirectional association as `BibDatabaseContext` already depends on `DBMSSynchronizer`. This would hinder the tests etc.

Now I extracted the method `setData(...)` from the constructor to achieve the goal.